llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7109,6 +7109,47 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
     }
   }
 
+  {
+    Value *X;
+    Constant *C;
+    if (match(Op0, m_OneUse(m_Intrinsic<Intrinsic::abs>(m_Value(X),
+                                                        m_Constant(C)))) &&
+        match(Op1, m_Specific(X))) {
+      auto *NullValue = Constant::getNullValue(X->getType());
+      auto *MinVal = ConstantInt::get(
+          X->getType(),
+          APInt::getSignedMinValue(X->getType()->getScalarSizeInBits()));
+      bool IsIntMinPosion = C->isZeroValue();
+      switch (Pred) {
+      case CmpInst::ICMP_ULE:
+      case CmpInst::ICMP_SGE:
+        return replaceInstUsesWith(I, ConstantInt::getTrue(I.getType()));
+      case CmpInst::ICMP_UGT:
+      case CmpInst::ICMP_SLT:
+        return replaceInstUsesWith(I, ConstantInt::getFalse(I.getType()));
+      case CmpInst::ICMP_UGE:
+      case CmpInst::ICMP_SLE:
+      case CmpInst::ICMP_EQ: {
+        auto *SGE = Builder.CreateICmpSGE(X, NullValue);
```

Comment:
```suggestion
        auto *SGT = Builder.CreateICmpSGT(X, AllOnesValue);
```
Canonicalize `sge X, 0 -> sgt X, -1`.